jenkins_pipelines/environments/common/pipeline-build-validation-cleanup.groovy

```diff
@@ -0,0 +1,249 @@
+        }
+
+        try {
+            stage('Clone terracumber, susemanager-ci and sumaform') {
+
+                // Prevent rebuild option
+                if (currentBuild.getBuildCauses().toString().contains("RebuildCause")) {
+                    error "Rebuild is blocked for this job."
+                }
+
+                // Create a directory for  to place the directory with the build results (if it does not exist)
+                sh "mkdir -p ${resultdir}"
+                git url: params.terracumber_gitrepo, branch: params.terracumber_ref
+                dir("susemanager-ci") {
+                    checkout scm
+                }
+
+                // Clone sumaform
+                sh "set +x; source /home/jenkins/.credentials set -x; ./terracumber-cli ${commonParams} --gitrepo ${params.sumaform_gitrepo} --gitref ${params.sumaform_ref} --runstep gitsync"
+
+                // Restore Terraform states from artifacts
+                if (params.use_previous_terraform_state) {
+                    copyArtifacts projectName: currentBuild.projectName, selector: specific("${currentBuild.previousBuild.number}")
+                }
+
+                // Set product version
+                if (params.targeted_project.contains("5.0")) {
+                    product_version = '5.0'
+                } else if (params.targeted_project.contains("4.3")) {
+                    product_version = '4.3'
+                } else if (params.targeted_project.contains("uyuni")) {
+                    product_version = 'uyuni'
+                }
+                else {
+                    // Use the `error` step instead of `throw`
+                    error("Error: targeted_project must contain either '5.0', '4.3' or uyuni.")
+                }
+            }
+
+            stage('Confirm Environment Cleanup') {
+                // Ask the user what environment they are cleaning, ensuring the answer matches params.targeted_project
+                def environmentChoice = input(
+                        message: 'What environment are you cleaning?',
+                        parameters: [
+                                string(name: 'Environment_Name', description: 'Enter the name of the environment you are cleaning.')
+                        ]
+                )
+
+                // Validate that the user entered the correct environment
+                if (environmentChoice != params.targeted_project) {
+                    error("The environment name entered does not match the targeted project. Aborting pipeline.")
+                }
+            }
+
+            stage("Copy terraform files from ${params.targeted_project}"){
+                // Copy tfstate and terraform directory to the result directory
+                sh """
+                    cp ${targetedTfStateFile} ${localTfStateFile}
+                    cp -r ${targetedTerraformDirPath}.terraform ${localSumaformDirPath}
+                """
```

Comment:
I wonder if we can achieve something similar with `copyArtifacts`. Maybe that would look nicer. But ok, just a matter of taste.

Reply:
I would not address that here. I can create an issue to test copyArtifacts. https://github.com/SUSE/spacewalk/issues/26011